mediastore/src/main/java/com/google/modernstorage/mediastore/MediaStoreRepository.kt

```diff
@@ -0,0 +1,415 @@
+                appContext,
+                Manifest.permission.WRITE_EXTERNAL_STORAGE
+            ) == PackageManager.PERMISSION_GRANTED
+        }
+    }
+
+    /**
+     * Get the [MediaStore] Image collection based on the provided [StorageLocation].
+     *
+     * @param location Location representing a storage volume.
+     */
+    private fun getImageCollection(location: StorageLocation): Uri {
+        return if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.Q) {
+            when (location) {
+                Internal -> MediaStore.Images.Media.getContentUri(MediaStore.VOLUME_INTERNAL)
+                SharedPrimary -> MediaStore.Images.Media.getContentUri(MediaStore.VOLUME_EXTERNAL_PRIMARY)
+            }
+        } else {
+            when (location) {
+                Internal -> MediaStore.Images.Media.INTERNAL_CONTENT_URI
+                SharedPrimary -> MediaStore.Images.Media.EXTERNAL_CONTENT_URI
+            }
+        }
+    }
+
+    /**
+     * Get the [MediaStore] Video collection based on the provided [StorageLocation].
+     *
+     * @param location Location representing a storage volume.
+     */
+    private fun getVideoCollection(location: StorageLocation): Uri {
+        return if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.Q) {
+            when (location) {
+                Internal -> MediaStore.Video.Media.getContentUri(MediaStore.VOLUME_INTERNAL)
+                SharedPrimary -> MediaStore.Video.Media.getContentUri(MediaStore.VOLUME_EXTERNAL_PRIMARY)
+            }
+        } else {
+            when (location) {
+                Internal -> MediaStore.Video.Media.INTERNAL_CONTENT_URI
+                SharedPrimary -> MediaStore.Video.Media.EXTERNAL_CONTENT_URI
+            }
+        }
+    }
+
+    /**
+     * Get the [MediaStore] Audio collection based on the provided [StorageLocation].
+     *
+     * @param location Location representing a storage volume.
+     */
+    private fun getAudioCollection(location: StorageLocation): Uri {
+        return if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.Q) {
+            when (location) {
+                Internal -> MediaStore.Audio.Media.getContentUri(MediaStore.VOLUME_INTERNAL)
+                SharedPrimary -> MediaStore.Audio.Media.getContentUri(MediaStore.VOLUME_EXTERNAL_PRIMARY)
+            }
+        } else {
+            when (location) {
+                Internal -> MediaStore.Audio.Media.INTERNAL_CONTENT_URI
+                SharedPrimary -> MediaStore.Audio.Media.EXTERNAL_CONTENT_URI
+            }
+        }
+    }
+
+    /**
+     * Get display name column in [MediaStore] based on the provided [FileType].
+     *
+     * @param type type of the media file.
+     */
+    private fun getDisplayNameColumn(type: FileType) = when (type) {
+        FileType.IMAGE -> MediaStore.Images.ImageColumns.DISPLAY_NAME
+        FileType.AUDIO -> MediaStore.Audio.AudioColumns.DISPLAY_NAME
+        FileType.VIDEO -> MediaStore.Video.VideoColumns.DISPLAY_NAME
```

Comment:
These all extend `MediaColumns`, which is where `DISPLAY_NAME`'s constant is declared. (And it certainly looks like every set of columns extends `MediaColumns`, including Downloads and Files?